### DISCUSSION TITLE
Assistance in including giscus for comment system

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#3221] [Q&A] [answered]
In the [Material for MkDocs 8 - Beta](https://github.com/squidfunk/mkdocs-material/issues/3219) issue was it mentioned, that Disqus has been removed due to its rather drastic changes in the past years.

While it's sad to lose this feature (at least natively) can I understand the reasons behind the removal of the native support.
I was now searching for an alternative and was thinking if a system based on GitHub's Discussions exists... Some small research later was I found happy to discover something called [giscus](https://github.com/giscus/giscus) which seems to be exactly what I am looking for.

There are currently two minor issues tho that I would need help with to get this hopefully working:

1. How can I integrate giscus into my MkDocs pages to display and allow comments? It uses a script implementation, so would that be enough to add through theme extension?
2. How could I disable this for particular pages? I don't want to have comments for particular pages like the main home page, so having a way to disable this would be good. Does Material support that through theme extensions?

And another unrelated question is, would it be considered to include a possible native support for giscus in the future?
It seems like a nice little comment system solution compared to Disqus and being open-source and free to use also seems like a good reason to support it.

**Comment by Andre601** [comment, marked as answer]
I managed to make it work by simply replacing the current script part in my disqus override with the one giscus gives me after configuring it.

Here's an example of my current `disqus.html` override:
```html
{% set disqus = config.extra.disqus %}
{% if page and page.meta and page.meta.disqus is string %}
  {% set disqus = page.meta.disqus %}
{% endif %}

<!-- Disqus integration -->
{% if not page.is_homepage and disqus %}
  <h2 id="__comments">
    <span class="twemoji">
      {% include ".icons/octicons/comment-discussion-24.svg" %}
    </span>
    {{ lang.t("meta.comments") }}
  </h2>
  <div id="disqus_thread"></div>
  <script src="https://giscus.app/client.js"
        data-repo="Andre601/blog"
        data-repo-id="MDEwOlJlcG9zaXRvcnkzMjMxMjM0MjQ="
        data-category="Comments"
        data-category-id="DIC_kwDOE0J44M4B_2e-"
        data-mapping="title"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-theme="dark_dimmed"
        data-lang="en"
        crossorigin="anonymous"
        async>
  </script>
{% endif %}
```
Of course, this for now only delays the day at which I have to find a new way to include comments in the MkDocs pages since the override won't function anymore unless squidfunk will implement something... But that's something to worry about for the future...

**Reply by squidfunk** [reply]
~~The new documentation page on "Adding a comment system" on the [`features/insiders-ghost-pepper`](https://github.com/squidfunk/mkdocs-material/pull/3218) branch includes a guide that lists exactly this information – i.e., what you need to do to add it back.~~ Nevermind – I read "Disqus", not "Giscus". We could definitely switch to this for the better!

Related commit: https://github.com/squidfunk/mkdocs-material/pull/3218/commits/f998d1e499e4ed4c45605000dd379d793655add2
